students/shawn_degraw/session03/strformat_lab.py

```diff
@@ -0,0 +1,56 @@
+from decimal import *
+
+#Task One
+print("Task One")
+print("File_{:03d} :  {:.2f}, {:.2e}, {:.3g}".format(2, 123.4567, 10000, 12345.67))
+
+#Task Two
+print("\nTask Two")
+value1=2
+value2=123.4567
+value3=10000
+value4=12345.67
+#?? trying to get significance of 3
+getcontext().prec = 3 
+#??not sure about 3 significant figures for value4
+print(f"File_{str(value1).zfill(3)} :  {round(value2,2)}, {Decimal(value3):.2e}, {Decimal(value4):.2e}")
+
+#Task Three
+#??No commas? 
+print("\nTask Three")
+def formatter(in_tuple):
+    form_string = f'the {str(len(in_tuple))} numbers are:  {("{:2}" * len(in_tuple))}'
```

Comment:
Minor: Nice construction of a variable-length formatting string at the end! Regarding the first part, I wonder if you had to use `str(...)`? Did it not work without it? I think it should have worked, and you may want to avoid unnecessary `str` construction (which is automatically done by the Python runtime for any f-string anyway).